urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/codegen/Method.java

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2009 Hao Xu
+ertranne@hotmail.com
+
+This file is part of EriLex.
+
+EriLex is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.codegen;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author ertri
+ */
+public class Method {
+
+    public String annotation;
+    public boolean cons;
+    public boolean abs;
+    public Type rType;
+    public String name;
+    public List<String> typeParams = new ArrayList<String>();
+    public List<Variable> params = new ArrayList<Variable>();
+    public List<Statement> mBody = new ArrayList<Statement>();
+    public boolean isStatic;
+
+    public Method(String name) {
+        this.cons = true;
+        this.name = name;
+    }
+
+    public Method(Type rType, String name) {
+        cons = false;
+        this.rType = rType;
+        this.name = name;
+    }
+
+    public Method(Type rType, String name, Type pType, String pName, Statement e) {
+        this(rType, name);
+        this.params.add(new Variable(pType, pName));
+        this.mBody.add(e);
+    }
+
+    public void codeGen(Writer w, String indent) throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 76-83, lines 92-100, lines 102-110 into separate methods.